### PR TITLE
A docker plugin to manage your containers

### PR DESCRIPTION
**Pull request by subutux**
## Docker plugin
A plugin to manage your docker containers (start, stop)
This uses the docker python module (https://docker-py.readthedocs.io)
Screenshot from 2019-03-11 13-48-32
trigger: `d `

### Actions:
* **toggle container** Using the default action (detects if the container is running or not)

### CODE REVIEW

**Comment by hulloanson** [comment]
The extension directory name `docker` conflicts with `docker` installed on python library directories. Albert tried to load the latter one instead. Can you change the directory name?

File: docker/__init__.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            for con in containers
+            if name in con.name.upper()]
+
+
+def handleQuery(query):
+    global conn
+    if query.isValid and query.isTriggered:
+        if error:
+            return error
+        return Filter(query.string, conn.containers.list(all=True))
```
Comment (hulloanson) [inline docker/__init__.py]: Can you catch connection errors? Sometimes when we don't have docker on or we have forgotten to add ourselves to the `docker` group after fresh installation, they just happen. I found these two while testing:

1. `ConnectionRefusedError` when docker is not running
2. Permission denied when I'm not in the `docker` group.

File: docker/__init__.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+__trigger__ = "d "
+__author__ = "subutux <subutux@gmail.com>"
+__dependencies__ = []
+
+error = None
+
+try:
+    import docker
+except ImportError:
+    error = [Item(id="docker-error",
```
Comment (idkCpp) [inline docker/__init__.py]: This seems like a fatal error. 
You are encouraged to throw this (or rethrow another if you believe that to be more descriptive) exception as the plugin will then be disabled in the plugin-manager.
I see no point in having the plugin enabled and returning an error every time.

**Comment by idkCpp** [comment]
I'm sorry to have messed this up. I don't get how this thing works...

**Comment by ManuelSchneid3r** [comment]
> The extension directory name `docker` conflicts with `docker` installed on python library directories. Albert tried to load the latter one instead. Can you change the directory name?

do you know if this is somehow avoidable? or should we prefix the extension names i.e. the modules?

**Comment by ManuelSchneid3r** [comment]
Sorry for letting you wait so long.

Please send the PR to the Python repo. The community approach did not work out.

Further I changed the policy considering plugins. If you want to be the active maintainer (own repo, own issuetracker, etc.) of the plugin, please place it in a separate repository and make this repo a git submodule of the Python plugin repo. If you do not want to be a maintainer, well, you can still send a PR to the Python plugin repo (no submodule in this case), but don't expect a merge. It depends on the demand and maintenance workload the plugin has.

**Pull request closed without merging**